### ISSUE TITLE
Ensure XnioSSL and XnioWorker resources are created only once in multithreaded environment (SimplePool v2) (1.6.x)

### ISSUE THREAD

**Issue by miklish** [networknt/light-4j#2053]
**NOTE:** This solution should not be merged until after PR https://github.com/networknt/light-4j/pull/2042 is merged

Ensure that XnioSsl, SSL Contexts, and XnioWorkers are only created once in SimpleUndertowConnectionMaker.

New PR: https://github.com/networknt/light-4j/pull/2054 (NOTE: **do not** merge before https://github.com/networknt/light-4j/pull/2042 has been merged)

**Comment by stevehu** [comment]
* Strict Resource Initialization: Updated SimpleClientConnectionMaker.java to use double-checked locking with synchronized blocks for getWorker() and getSSL().
* Performance & Reliability: This ensures that XnioWorker and XnioSsl (including SSL context creation and keystore loading) are instantiated exactly once, even if multiple threads request a connection simultaneously during startup.
